Improve Nest troubleshooting readability (#38522)
home-assistant/home-assistant.github.io: source/_integrations/nest.markdown

{% endnote %}

> ## Known Limitations

Reviewer: ```suggestion
## Known limitations
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization), also in headings.

> ### Google Account Types

Reviewer: ```suggestion
### Google account types
```

> There are limitations about which Google Accounts can use the SDM API. See the [Device Access Registration](https://developers.google.com/nest/device-access/registration) documentation for details.

Reviewer: ```suggestion
There are limitations to which Google accounts can use the SDM API. See the [Device Access Registration](https://developers.google.com/nest/device-access/registration) documentation for details.
```

> The primary limitations are:

Reviewer: ```suggestion
The primary limitations are the following:
```

> - Google Workspace accounts are not supported. Only consumer accounts (e.g. gmail.com) can be used.

Reviewer: ```suggestion
- Google Workspace accounts are not supported. Only consumer accounts (for example, gmail.com) can be used.
```

> - Once a Google Account is associated with your Device Access Project, it cannot be changed Be sure you are signed in to the correct Google Account before continuing.

Reviewer: ```suggestion
- Once a Google Account is associated with your Device Access Project, it cannot be changed. Be sure you are signed in to the correct Google Account before continuing.
```

Keep in mind, the US$5 registration fee is non-refundable.

> ### Google Home App Migration and Cameras

Reviewer: ```suggestion
### Google Home App migration and cameras
```

Migrating your cameras to the Google Home App converts cameras from RTSP to WebRTC, which results in the loss of snapshot functionality. Additionally, there is a known issue where media is not currently published for these events; this has been reported to the Nest SDM team and is an unexpected side effect of their migration.

### Temperature sensors partially supported

> Additional Nest Temperature Sensors are not supported by the SDM API. The Temperature reported by the API will be pulled from whichever device is currently configured as the Active Sensor, which can be adjusted via manual selection or the schedule offered in the Nest App. If multiple sensors are available, only the temperature from the active sensor will be displayed.

Reviewer: ```suggestion
Additional Nest temperature sensors are not supported by the SDM API. The temperature reported by the API will be pulled from whichever device is currently configured as the Active Sensor, which can be adjusted via manual selection or the schedule offered in the Nest App. If multiple sensors are available, only the temperature from the active sensor will be displayed.
```

## Troubleshooting

> ### Can't Link Google Account

Reviewer: ```suggestion
### Can't link Google account
```

#### Symptom: Can’t link to [Project Name]: Please contact [Project Name] if the issue persists

##### Description

The error *Can’t link to [Project Name]* typically means that the *OAuth Client ID* used is
mismatched in Home Assistant [Application Credentials](/integrations/application_credentials/).

##### Resolution

The simplest approach is to remove Application Credentials from Home Assistant and then verify
that the *OAuth Client ID* used must be consistent across these three places:

> - [Google Cloud Console](https://console.cloud.google.com/apis/credentials) - See instructions above to create new Web Auth OAuth Credentials if needed

Reviewer: ```suggestion
- [Google Cloud Console](https://console.cloud.google.com/apis/credentials) - See instructions above to create new Web Auth OAuth Credentials if needed.
```

> - [Device Access Project](https://console.nest.google.com/device-access/project-list) - The OAuth Client ID for your Device Access Project must refer to the Web Auth OAuth Client ID in the Google Cloud Console

Reviewer: ```suggestion
- [Device Access Project](https://console.nest.google.com/device-access/project-list) - The OAuth Client ID for your Device Access Project must refer to the Web Auth OAuth Client ID in the Google Cloud Console.
```

> - Make sure you are using the same Google Account in the Device Access Console and Google Cloud Console e.g. double-check the photo and account name in the top right of the screen

Reviewer: ```suggestion
- Make sure you are using the same Google Account in the Device Access Console and Google Cloud Console. For example, double-check the photo and account name in the top right of the screen.
```

  ![Screenshot of success](/images/integrations/nest/application_credentials.png)

#### Symptom: No access to partner information or Information could not be retrieved

##### Description

The error *No access to partner information* or *Information could not be retrieved* shown
during the account linking process means that the Google Account used cannot access the Google Home.

##### Resolution

- You can organize your homes and devices in the Google Home App and [share homes and devices](https://support.google.com/googlenest/answer/9155535) across accounts. Ensure the account being used has access to the Home.
- If you formerly had a Nest account, ensure that it is migrated successfully to a Google Account. If your Google Home has multiple members, please note that the individual who initially set up the home must complete the migration of their Nest Account to a Google Account before you can establish a connection with Home Assistant.

#### Symptom: Error 400: redirect_uri_mismatch 

##### Description

> This error *Error 400: redirect_uri_mismatch * means that that your OAuth Client ID is not configured to match the *My Home Assistant* callback URL. Home Assistant's redirect URL behavior may have changed since you initially set this up!

Reviewer: ```suggestion
This error *Error 400: redirect_uri_mismatch* means that your OAuth Client ID is not configured to match the *My Home Assistant* callback URL. Home Assistant's redirect URL behavior may have changed since you initially set this up!
```

##### Resolution

Confirm that your Nest device is visible within the Google Home App and Nest App under the same Home. If it is missing within Google Home, create a new dummy home on the Nest app, which triggers the synchronization process. (This is the workaround recommended by the Google support team). The dummy entry can be deleted once the Nest device is visible within the Google Home app.

#### Symptom: Error 403: access_denied or Access blocked: home-assistant.io has not completed Google's verification process

##### Description

The error *Error 403: access_denied* means that the OAuth Consent screen may be misconfigured,
either because it does not allow access to your Google Account or because you have entered
extra information that triggered Google's verification process. Google will require
verification when you add extra information to the branding page.

#### Resolution

Visit the OAuth [Verification Center](https://console.cloud.google.com/auth/verification) and
confirm the *Verification Status* is *Verification not required*. If verification
is required:

> 1. Navigate to the [Branding](https://console.cloud.google.com/auth/branding) page

Reviewer: ```suggestion
1. Navigate to the [Branding](https://console.cloud.google.com/auth/branding) page.
```

2. Remove additional fields that are not required and Save. See the set up instruction above and
make sure not to enter extra fields.
3. Go back to the Verification Center and confirm the status is correct.

Additionally you need to make sure the Audience configuration is correct by following these steps:

> 1. Visit the OAuth [Audience](https://console.cloud.google.com/auth/audience) page

Reviewer: ```suggestion
1. Visit the OAuth [Audience](https://console.cloud.google.com/auth/audience) page.
```

2. Make sure the account is set to *In production*.

You may now repeat the integration setup and account linking steps.

#### Symptom: Error: invalid_client no application name

##### Description

The error *Error: invalid_client no application name* means the OAuth Consent screen has not been

> fully configured for the project and needs additional information

Reviewer: ```suggestion
fully configured for the project and needs additional information.
```

#### Resolution

Visit the [OAuth Consent Screen](https://console.developers.google.com/apis/credentials/consent) and
enter the required fields (App Name, Support Email, Developer Email) and leave everything else as default.

> ### Can't find Devices

Reviewer: ```suggestion
### Can't find devices
```

#### Symptom: Devices do not appear in Home Assistant

##### Description

This typically means that Home Assistant does not have access to the device because it was not returned
from the SDM API.

##### Resolution

You can add or remove devices and permissions granted to Home Assistant in the Nest [Partner Connections Manager](https://nestservices.google.com/partnerconnections). Reload the Nest integration to make new devices available. See the [SDM API Troubleshooting](https://developers.google.com/nest/device-access/authorize#modify_account_permissions) documentation for more details.

#### Symptom: Thermostats do not appear in Home Assistant or are unavailable

There have been reports that Thermostats may not appear or are unavailable due to a bug in the SDM API. A common fix to get the API to work again is to try these steps:

- Restart the Thermostat device. See [How to restart or reset a Nest thermostat](https://support.google.com/googlenest/answer/9247296) for more details.
- In the official Nest app or on https://home.nest.com: Move the Thermostat to a different or fake/temporary room.

> - Reload the integration in Home Assistant:  Navigate to **Configuration** then **Devices & services**, click `...` next to *Nest* and choose **Reload**.

Reviewer: ```suggestion
- Reload the integration in Home Assistant:  Navigate to {% my integrations title="**Settings** > **Devices & services**" %}, select {% icon "mdi:dots-vertical" %} next to *Nest* and choose **Reload**.
```
Please ignore this comment if I got this wrong.

Reviewer 2: This seems correct.

#### Symptom: Devices do not appear when the API is disabled

##### Description

The SDM API may not return devices for the authorized account when the SDM API is disabled.

##### Resolution

Double-check that GCP is configured correctly and [Enable the API](https://developers.google.com/nest/device-access/get-started#set_up_google_cloud_platform) and authorize at least one device in the OAuth setup flow. If you have trouble here, then you may want to walk through the Google instructions and issue commands directly against the API until you successfully get back the devices.

> ### Authentication Problems

Reviewer: ```suggestion
### Authentication problems
```

- This most likely reason is the *OAuth Consent Screen* is set to *Testing* by default which expires the token after 7 days.
- Follow the steps above to set it to *Production* to resolve this and reauthorize your integration one more time to get a new token.
- You may also see this as the error message *invalid_grant: Token has been expired or revoked*.
- See [Google Identity: Refresh token expiration](https://developers.google.com/identity/protocols/oauth2#expiration) for more reasons on why your token may have expired.

### Integration set up failure

#### Symptom: Configuration error: Failed to create subscriber `subscription/name` was not found

##### Description

The integration fails to start because it attempts to create a subscriber with a subscription
name that is not found in your Google Account. By default, Google pub/sub subscriptions will be deleted after 31 days of inactivity ([reference](https://cloud.google.com/knowledge/kb/pub-sub-subscriptions-disappeared-without-any-deletion-logs-000004170)). If this happens, then the integration will fail, and you will see the preceding log line in your Home Assistant logs.

##### Resolution.

To repair the subscriber:

>   1. Go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) and re-enable the Pub/Sub Topic.

Reviewer: ```suggestion
  1. Go to the [Device Access Console](https://console.nest.google.com/device-access/project-list) and re-enable the Pub/Sub topic.
```

  2. The Nest integration may need to be recreated to pick up the new Pub/Sub topic.
  3. (Optional) To prevent future expirations, go to your [Google Cloud Console Pub/Sub subscription page](https://console.cloud.google.com/cloudpubsub/subscription/list) and edit the Pub/Sub subscription created by Nest to not expire by default.

### Nest integration data issues

#### Symptom: Not receiving camera motion and person events

The Nest integration subscribes a Google Pub/sub subscription to listen for camera motion
or person events. The settings in the Google Home app also control which events
are published, so if they are not set properly you may not receive events.

#### Resolution

> - Verify that you have allowed Home Assistant to access camera streams, and permissions are correctly set in [Partner Connections Manager](https://nestservices.google.com/partnerconnections))

Reviewer: ```suggestion
- Verify that you have allowed Home Assistant to access camera streams, and permissions are correctly set in [Partner Connections Manager](https://nestservices.google.com/partnerconnections).
```
